EventIterator.hpp

```diff
@@ -31,14 +31,87 @@
 #include <iterator>
 #include <limits>
 
+/** abstract base class for an iterator over the elements contained in a data block */
+class DataBlockBaseIterator{
+protected:
+  const caen::ReadoutBuffer& buffer;
+  uint32_t* ptr;
+public:
+  DataBlockBaseIterator(const caen::ReadoutBuffer& b)
+    : buffer(b)
+    , ptr((uint32_t*)buffer.data) {}
+  virtual uint16_t group() { return 0; }
+  virtual void* end() const { return buffer.end(); }
+  virtual DataBlockBaseIterator& operator++() { return *this; }
+  bool operator==(const DataBlockBaseIterator& other) const { return ptr == other.ptr; }
+  bool operator!=(const DataBlockBaseIterator& other) const { return (ptr != other.ptr); }
+  bool operator> (const DataBlockBaseIterator& other) const { return ptr > other.ptr; }
+  bool operator< (const DataBlockBaseIterator& other) const { return ptr < other.ptr; }
+  bool operator>=(const DataBlockBaseIterator& other) const { return ptr >= other.ptr; }
+  bool operator<=(const DataBlockBaseIterator& other) const { return ptr <= other.ptr; }
+  bool operator==(const void* other) const { return ptr == other; }
+  bool operator!=(const void* other) const { return ptr != other; }
+  bool operator> (const void* other) const { return ptr > other; }
+  bool operator< (const void* other) const { return ptr < other; }
```

Comment:
I can't believe that all of these are necessary. :-)

Reply:
I would say that you are correct. [It is not.](https://en.cppreference.com/w/cpp/language/operators)

Reply:
No, I don't think so either. I guess they were originally put in for completeness sake.